streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinMerger.java

```diff
@@ -117,13 +120,16 @@ public void init(final ProcessorContext<K, Change<V>> context) {
         @Override
         public void process(final Record<K, Change<V>> record) {
             if (queryableName != null) {
-                store.put(record.key(), record.value().newValue, record.timestamp());
-                tupleForwarder.maybeForward(record);
+                final long putReturnCode = store.put(record.key(), record.value().newValue, record.timestamp());
+                // if not put to store, do not forward downstream either
```

Comment:
Could this ever happen?

Reply:
Accidentally responded above in https://github.com/apache/kafka/pull/13564#discussion_r1171930596, instead of here.